src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -62,9 +67,10 @@ export class RenderLineInput {
 		tabSize: number,
 		spaceWidth: number,
 		stopRenderingLineAfter: number,
-		renderWhitespace: 'none' | 'boundary' | 'all',
+		renderWhitespace: 'none' | 'boundary' | 'selection' | 'all',
 		renderControlCharacters: boolean,
-		fontLigatures: boolean
+		fontLigatures: boolean,
+		selectionsOnLine: Range[] | undefined // NOTE this will only be defined when renderWhitespace is 'selection'
```

Comment:
Also, I think `Range` is the wrong class to use here because the `Range` is not guaranteed to be on the current line number. This rendering code is line number agnostic to allow for optimizations in cases like pressing Enter... so I would like to please remove the `lineNumber` "knowledge" out of here. I suggest to create something like:

```ts
class LineRange { 
	public readonly startIndex; // or startOffset
	public readonly length; // or perhaps endIndex, endOffset
}
```

You can create a new class that only contains the info captured in `startColumn` and `endColumn`. I also tried to leave the `column` world behind in these methods and talk about character offsets so that's why I recommend using `offset` or `index` and not `column` in the above suggested class. When I wrote this code I was a big fan of calling everything index, in the meantime I have a better preference towards offset when talking about a position inside a string and a preference towards index when talking about a position inside an array...